packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -31,6 +33,22 @@ const Search = props => {
         indexName: algolia.indexName,
         inputSelector: '#search_input_react',
         algoliaOptions: algolia.algoliaOptions,
+        // Override algolia's default selection event, allowing us to do client-side
+        // navigation and avoiding a full page refresh.
+        handleSelected: (_input, _event, suggestion) => {
+          // Use an anchor tag to parse the absolute url into a relative url
+          // Alternatively, we can use new URL(suggestion.url) but its not supported in IE
+          const a = document.createElement('a');
+          a.href = suggestion.url;
+
+          // Algolia use closest parent element id #__docusaurus when a h1 page title does not have an id
+          // So, we can safely remove it. See https://github.com/facebook/docusaurus/issues/1828 for more details.
+          const routePath =
+            `#__docusaurus` === a.hash
```

Comment:
I didnt want to do replace() because thats not supported in older browser and will have to be polyfilled :v